### ISSUE TITLE
Functional categorisation of functions.

### ISSUE THREAD

**Issue by aron0093** [siddharthst/CoreFunctions#2]
Creating different files representing specific functional categories to store the functions will make them more accessible. I propose this be done by both analytical purpose aswell as the nature of the function itself. 

For example, interactive_visualisation.py might contain functions that result in interactive plots.

@siddharthst What do you think?

**Comment by siddharthst** [comment]
I agree. However, I believe it should be based on the methods themselves instead of broad categories like interactive visualisation. An example of this would be PCA class - which will do the PCA given pandas data frame and along with that, will also plot the accompanying visualisations. 
Typical case: http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/112-pca-principal-component-analysis-essentials/

**Comment by aron0093** [comment]
I agree that we must create classes corresponding to techniques like PCA which input data as pandas frames and the associated object has the processed data as well as data for plotting. It will also have the option to generate the plots. 

However these building blocks also need to be combined in terms of the tasks that a biologist needs to perform. So we need also create a dashboard functionality that can take any of the building blocks. E.g. a dashboard that has differential expression analysis, pathway enrichment etc. could be a useful tool to characterise clusters in exploring single cell rnaseq data. 

I think developing these pre-built and customisable dashboards might be the key differentiator compared to other packages like scanpy or more general libraries like seaborn.
